### PR TITLE
100% coverage on skills/error

### LINKED ISSUE #306: Skill error handlers.
**Describe the bug**
The error.handlers.py's function send_unsupported_skill asks for a protocol to pass it to the logger and print out a warning. The protocols we are using have no id. 

Example: OEFMessage.protocol_id  is str. 

If we pass this to the send_unsupported_skill(envelope, OEFMesasge.protocol_id)
returns error since the protocol has no id.

**Expected behaviour**
Can retrieve the protocol id from the envelope

**Desktop (please complete the following information):**
 - OS: [Mac.OS]

### PR DESCRIPTION
**Pull request by Totoual**
## Proposed changes

100% coverage on error_skill 

## Fixes

#306 
Changes the handler to retrieve the protocol.id from the envelop.
## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments
None

### CODE REVIEW

File: aea/connections/local/connection.py [hunk 1]
```diff
@@ -99,6 +99,8 @@ def handle_oef_message(self, envelope: Envelope) -> None:
         oef_type = OEFMessage.Type(oef_message.get("type"))
         if oef_type == OEFMessage.Type.REGISTER_SERVICE:
             self.register_service(sender, cast(Description, oef_message.get("service_description")))
+        elif oef_type == OEFMessage.Type.REGISTER_AGENT:
+            self.register_service(sender, cast(Description, oef_message.get("agent_description")))
```
Comment (DavidMinarsch) [inline aea/connections/local/connection.py]: Did you mean `register_agent`?

File: aea/protocols/oef/message.py [hunk 2]
```diff
@@ -85,6 +89,14 @@ def check_consistency(self) -> bool:
                 service_id = self.get("service_id")
                 assert isinstance(service_description, Description)
                 assert isinstance(service_id, str)
+            elif oef_type == OEFMessage.Type.REGISTER_AGENT:
+                assert self.is_set("id")
+                assert self.is_set("agent_description")
+                assert self.is_set("agent_id")
```
Comment (DavidMinarsch) [inline aea/protocols/oef/message.py]: What is `agent_id`?

File: tests/test_protocols/test_oef_message.py [hunk 3]
```diff
@@ -34,6 +34,10 @@ def test_oef_type_string_value():
         "The string representation must be register_service"
     assert str(OEFMessage.Type.UNREGISTER_SERVICE) == "unregister_service",\
         "The string representation must be unregister_service"
+    assert str(OEFMessage.Type.REGISTER_AGENT) == "register_agent",\
+        "The string representation must be register_service"
```
Comment (DavidMinarsch) [inline tests/test_protocols/test_oef_message.py]: change to `register_agent`

File: tests/test_protocols/test_oef_message.py [hunk 4]
```diff
@@ -34,6 +34,10 @@ def test_oef_type_string_value():
...
+    assert str(OEFMessage.Type.UNREGISTER_AGENT) == "unregister_agent",\
+        "The string representation must be unregister_service"
```
Comment (DavidMinarsch) [inline tests/test_protocols/test_oef_message.py]: change to `unregister_agent`

File: tests/test_skills/test_error/test_error_skill_handlers.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        cls.wallet = Wallet({'default': None})
+        cls.mailbox1 = MailBox(OEFLocalConnection(cls.wallet.public_keys['default'], cls.node))
+        cls.my_aea = AEA("Agent0", cls.mailbox1, cls.wallet, directory=str(Path(CUR_PATH, "data/dummy_aea")))
+
+        cls.skill_context = SkillContext(cls.my_aea.context)
+        cls.my_error_handler = ErrorHandler(skill_context=cls.skill_context)
+
+    def test_error_handler_setup(self):
+        """Test the setup."""
+        self.my_error_handler.setup()
```
Comment (DavidMinarsch) [inline tests/test_skills/test_error/test_error_skill_handlers.py]: These aren't really testing anything. You are just calling these methods on the error handler.

Comment (Totoual) [reply tests/test_skills/test_error/test_error_skill_handlers.py]: yes it is in-order to call the `pass` on the handler. Didn't want to add` # pragma: no cover.`

**Pull request merged**